caffeine/src/main/scala/scalacache/caffeine/CaffeineCache.scala

```diff
@@ -0,0 +1,89 @@
+package scalacache.caffeine
+
+import scalacache.{ LoggingSupport, Cache, Entry }
+import com.github.benmanes.caffeine.cache.{ Cache => CCache, Caffeine }
+
+import scala.concurrent.duration.Duration
+import org.joda.time.DateTime
+import com.typesafe.scalalogging.StrictLogging
+import scala.concurrent.Future
+
+/*
+ * Thin wrapper around Caffeine.
+ * Since Caffeine is in-memory and non-blocking,
+ * all operations are performed synchronously, i.e. ExecutionContext is not needed.
```

Comment:
Both Guava and Caffeine are blocking during the compute. You could use Caffeine's `AsyncLoadingCache` with [scala-java8-compat](https://github.com/scala/scala-java8-compat) to convert to Scala Futures. However, async is incompatible with some features so you might want that to be an alternative adapter.